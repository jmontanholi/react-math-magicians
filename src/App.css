body {
  background-color: rgb(217, 239, 255);
}

#root {
  display: flex;
  justify-content: center;
}
