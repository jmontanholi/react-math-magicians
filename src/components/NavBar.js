import React from 'react';
import { Link } from 'react-router-dom';
import style from './NavBar.module.css';

const NavBar = () => (
  <nav className={style.nav}>
    <div className={style.titleDiv}>
      <h1 className={style.title}>
        Math Magicians
      </h1>
    </div>
    <div className={style.linksDiv}>
      <Link data-testid="linkHome" className={style.link} to="/">Home</Link>
      <Link data-testid="linkCalc" className={`${style.link} ${style.link2}`} to="/calculator">Calculator</Link>
      <Link data-testid="linkQuote" className={style.link} to="/quote">Quote</Link>
    </div>
  </nav>
);

export default NavBar;
