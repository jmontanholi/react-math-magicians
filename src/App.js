/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import './App.css';
import './componentsCss/calculatorGrid.css';
import Calculator from './components/calculator';

class App extends React.Component {
  render() {
    return (
      <div>
        <Calculator />
      </div>
    );
  }
}

export default App;
