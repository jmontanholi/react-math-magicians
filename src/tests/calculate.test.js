import React from 'react';
// import calculate from '../logic/calculate';
// import operate from '../logic/operate';
import { fireEvent, render, screen } from '@testing-library/react';
// import '@testing-library/jest-dom';
// import App from '../App';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';
import NavBar from '../components/NavBar';

describe('NavBar', () => {
  it('Renders the NavBar', () => {
    const nav = TestRenderer
      .create(
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(nav).toMatchSnapshot();
  });

  it('Renders calculator page', () => {
    render(<App />);

    fireEvent.click(screen.getByTestId('linkCalc'));

    const title = screen.getByTestId('mathTitle');
    expect(title.textContent).toBe("Let's do some Math");
  });

  it('Renders home page', () => {
    render(<App />);

    fireEvent.click(screen.getByTestId('linkHome'));

    const title = screen.getByTestId('homeTitle');
    expect(title.textContent).toBe('Welcome To Our Page!');
  });

  it('Renders quote page', () => {
    render(<App />);

    fireEvent.click(screen.getByTestId('linkQuote'));

    const title = screen.getByTestId('quoteTitle');
    expect(title.textContent).toBe('Quote of the day:');
  });
});
