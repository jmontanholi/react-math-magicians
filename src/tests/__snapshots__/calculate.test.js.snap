// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Renders the NavBar 1`] = `
<nav
  className="nav"
>
  <div
    className="titleDiv"
  >
    <h1
      className="title"
    >
      Math Magicians
    </h1>
  </div>
  <div
    className="linksDiv"
  >
    <a
      className="link"
      data-testid="linkHome"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="link link2"
      data-testid="linkCalc"
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      className="link"
      data-testid="linkQuote"
      href="/quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </div>
</nav>
`;
